v2/bigtable-cdc-to-hbase/src/main/java/com/google/cloud/teleport/v2/templates/BigtableToHbasePipeline.java

```diff
@@ -0,0 +1,282 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.teleport.metadata.Template;
+import com.google.cloud.teleport.metadata.TemplateCategory;
+import com.google.cloud.teleport.metadata.TemplateParameter;
+import com.google.cloud.teleport.v2.templates.transforms.ChangeStreamToRowMutations;
+import com.google.cloud.teleport.v2.templates.transforms.HbaseRowMutationIO;
+import com.google.cloud.teleport.v2.templates.utils.RowMutationsCoder;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.gcp.bigtable.BigtableIO;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.ExperimentalOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.client.RowMutations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Bigtable change stream pipeline to replicate changes to Hbase. Pipeline reads from a Bigtable
+ * change stream, converts change stream mutations to their nearest Hbase counterparts, and writes
+ * the resulting Hbase row mutations to Hbase.
+ *
+ * <p>In Bigtable, all writes to a single row on a single cluster will be streamed in order by
+ * commitTimestamp. This is only the case for instances with a single cluster, or if all writes to a
+ * given row happen on only one cluster in a replicated instance. This order is maintained in the
+ * Dataflow connector via key-ordered delivery, as the Dataflow connector emits ChangeStreamMutation
+ * records with the row key as the record key.
+ */
+@Template(
+    name = "bigtable-cdc-to-hbase",
+    category = TemplateCategory.STREAMING,
+    displayName = "Bigtable CDC to HBase Replicator",
+    description = "A streaming pipeline that replicates Bigtable change stream data to HBase",
+    optionsClass = BigtableToHbasePipeline.BigtableToHbasePipelineOptions.class,
+    flexContainerName = "bigtable-cdc-to-hbase",
+    contactInformation = "https://cloud.google.com/support")
+public class BigtableToHbasePipeline {
```

Comment:
Naming convention is typically {source}To{sink} (i.e. `BigtableToHbase`). I would remove the "Pipeline" suffix.

Reply:
Renamed to BigtableCdcToHbase